Makefile

```diff
@@ -272,6 +262,37 @@ debug-development: build-development-image ## Debug development environment
 		$(docker_development_image_repository):$(docker_image_version) \
 		/bin/bash
 
+format: build-development-image ## Format all of the source code with the rules in .clang-format
+
+	docker run \
+		--rm \
+		--volume="$(CURDIR):/app" \
+		--workdir=/app \
+		--user="$(shell id -u):$(shell id -g)" \
+		$(docker_development_image_repository):$(docker_image_version) \
+		clang-format -i -style=file:thirdparty/clang/.clang-format ${clang_format_sources_path}
+
+format-check: build-development-image ## Runs the clang-format tool to check the code against rules and formatting
+
+	docker run \
+		--rm \
+		--volume="$(CURDIR):/app" \
+		--workdir=/app \
+		--user="$(shell id -u):$(shell id -g)" \
+		"$(docker_development_image_repository):$(docker_image_version)" \
+		clang-format -Werror --dry-run -style=file:thirdparty/clang/.clang-format ${clang_format_sources_path}
+
+format-python: build-development-image  ## Runs the black format tool against python code
+
+	docker run \
+		--rm \
+		--privileged \
+		--volume="$(CURDIR):/app:delegated" \
+		--workdir=/app \
+		$(docker_development_image_repository):$(docker_image_version) \
+		/bin/bash -c "python3.11 -m black --line-length=90 bindings/python/"
+
```

Comment:
Use the `.PHONY` per target, instead of having a full list at the bottom to align with astro